tests/Database/DatabaseEloquentBuilderFindOrFailWithEnumTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Illuminate\Tests\Database;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Orchestra\Testbench\TestCase;
+use PHPUnit\Framework\Attributes\Test;
+
+class DatabaseEloquentBuilderFindOrFailWithEnumTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchema();
+    }
+
+    protected function tearDown(): void
+    {
+        Schema::drop('test_models');
+
+        parent::tearDown();
+    }
+
+    protected function getEnvironmentSetUp($app)
+    {
+        $app['config']->set('database.default', 'testing');
+    }
+
+    #[Test]
+    public function it_finds_existing_model_when_using_enum_id()
```

Comment:
We don't use `Test` attribute in the framework, prefix the method name with `test` and use studly case.